### PR TITLE
Fix .zip decompression when unzip is not installed

### PR DESCRIPTION
**Pull request by SirLynix**
tar doesn't handle .zip:
```
/usr/bin/tar -xf v5.2.3.zip -C source.tmp
/usr/bin/tar: This does not look like a tar archive
/usr/bin/tar: Skipping to next header
/usr/bin/tar: Exiting with failure status due to previous errors
error: @programdir/core/sandbox/modules/os.lua:372: execv(/usr/bin/tar -xf v5.2.3.zip -C source.tmp) failed(2)
stack traceback:
    [C]: in function 'error'
    [@programdir/core/base/os.lua:897]:
    [@programdir/core/sandbox/modules/os.lua:372]:
    [@programdir/core/sandbox/modules/os.lua:285]: in function 'vrunv'
    [@programdir/modules/utils/archive/extract.lua:90]: in function 'extract'
    [@programdir/modules/utils/archive/extract.lua:374]:
    [...modules/private/action/require/impl/actions/download.lua:177]: in function '_download'
    [...modules/private/action/require/impl/actions/download.lua:289]:

  => download https://github.com/assimp/assimp/archive/v5.2.3.zip .. failed
```

### CODE REVIEW

**Comment by waruqi** [comment]
it works on macOS

```console
$ tar -xvf sh_unix_v6-master.zip 
x sh_unix_v6-master/
x sh_unix_v6-master/LICENSE
x sh_unix_v6-master/README.md
x sh_unix_v6-master/sh.c
```

**Comment by SirLynix** [comment]
I updated it

**Comment by waruqi** [comment]
syntax error

**Comment by SirLynix** [comment]
My mistake, fixed

**Comment by SirLynix** [comment]
I also made it more resilient to extractor failure:
```
/usr/bin/tar -xf fmt-9.1.0.zip -C source.tmp
/usr/bin/tar: This does not look like a tar archive
/usr/bin/tar: Skipping to next header
/usr/bin/tar: Exiting with failure status due to previous errors
/usr/bin/7z x -y fmt-9.1.0.zip -osource.tmp

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=C.UTF-8,Utf16=on,HugeFiles=on,64 bits,32 CPUs AMD Ryzen 9 5950X 16-Core Processor             (A20F12),ASM,AES-NI)

Scanning the drive for archives:
1 file, 1262685 bytes (1234 KiB)

Extracting archive: fmt-9.1.0.zip
--
Path = fmt-9.1.0.zip
Type = zip
Physical Size = 1262685

Everything is Ok
```

File: xmake/modules/utils/archive/extract.lua [hunk 1]
```diff
@@ -371,7 +371,13 @@ end
 -- extract archive file using extractors
 function _extract(archivefile, outputdir, extension, extractors, opt)
     for _, extract in ipairs(extractors) do
-        if extract(archivefile, outputdir, extension, opt) then
+        local ok = try
+        {
+            function ()
+                return extract(archivefile, outputdir, extension, opt)
+            end
+        }
```
Comment (waruqi) [inline xmake/modules/utils/archive/extract.lua:379]:
```lua
        local ok = try {function () return extract(archivefile, outputdir, extension, opt) end}
```

**Pull request merged**